tests/std/tests/P0896R4_counted_iterator/test.cpp

```diff
@@ -0,0 +1,286 @@
+        // [counted.iter.const]
+        {
+            [[maybe_unused]] counted_iterator<Iter> defaultConstructed{};
+
+            counted_iterator<Iter> constructed(Iter{input}, iter_difference_t<Iter>{2});
+            counted_iterator<Iter> constructedEmpty{Iter{input}, iter_difference_t<Iter>{0}};
+
+            if constexpr (copyable<Iter>) { // counted_iterator only has const lvalue conversions
+                counted_iterator<ConstIter> constructedConversion{constructed};
+                constructedConversion = constructedEmpty;
+            }
+        }
+
+        // [counted.iter.access]
+        {
+            if constexpr (copy_constructible<Iter>) {
+                const counted_iterator<Iter> lvalueBase{Iter{input}, 2};
+                const same_as<Iter> auto base1 = lvalueBase.base();
+                assert(base1.peek() == input);
+            }
+
+            const same_as<Iter> auto base2 = counted_iterator<Iter>{Iter{input}, 2}.base();
+            assert(base2.peek() == input);
+
+            const same_as<iter_difference_t<Iter>> auto length = counted_iterator<Iter>{Iter{input}, 2}.count();
+            assert(length == 2);
+        }
+
+        // [counted.iter.elem]
+        {
+            counted_iterator<Iter> iter{Iter{input}, 2};
+            assert(*iter == 1);
+            if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                assert(addressof(*iter) == input);
+            }
+
+            const counted_iterator<Iter> constIter{Iter{input}, 2};
+            assert(*constIter == 1);
+            if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                assert(addressof(*constIter) == input);
+            }
+
+            if constexpr (random_access_iterator<Iter>) {
+                assert(iter[1] == 2);
+                if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                    assert(addressof(iter[1]) == input + 1);
+                }
+            }
+        }
+        // [counted.iter.nav]
+        {
+            { // pre increment
+                counted_iterator<Iter> iter{Iter{input}, 2};
+                assert(addressof(++iter) == addressof(iter));
+                assert(iter.count() == 1);
+                assert(*iter == 2);
+                if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                    assert(addressof(*iter) == input + 1);
+                }
+            }
+            if constexpr (forward_iterator<Iter>) { // post increment
+                counted_iterator<Iter> iter{Iter{input}, 2};
+                auto ref   = iter;
+                auto iter2 = iter++;
+                assert(iter2 == ref);
+                assert(iter.count() == 1);
+                assert(*iter == 2);
+                if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                    assert(addressof(*iter) == input + 1);
+                }
+            }
+            if constexpr (bidirectional_iterator<Iter>) {

```

Comment:
This is test code so I care significantly less about stylistic conventions, but I recommend having a newline between non-chained control flow; here you have `if constexpr` without any empty lines. Occurs below.

Reply:
Added a newline